src/libsyntax/source_map.rs

```diff
@@ -96,9 +96,7 @@ impl FileLoader for RealFileLoader {
     }
 
     fn read_file(&self, path: &Path) -> io::Result<String> {
-        let mut src = String::new();
-        fs::File::open(path)?.read_to_string(&mut src)?;
-        Ok(src)
+        fs::read_to_string(path)
```

Comment:
If we're going to do this I *kind* of feel like we might as well get rid of read_file entirely? It used to be better but seems strictly "worse" now

Reply:
I think we want to keep the filesystem operations trait-ified, right?

Reply:
Oh, yes, I missed that this was a trait implementation.